libraries/botbuilder-ai-orchestrator/src/orchestratorRecognizer.ts

```diff
@@ -257,7 +250,18 @@ export class OrchestratorRecognizer extends AdaptiveRecognizer implements Orches
             recognizerResult.intents.None = { score: 1.0 };
         }
 
-        await this.tryScoreEntities(text, recognizerResult);
+        if (this.externalEntityRecognizer) {
```

Comment:
Why should this be outside tryScoreEntities?  externalEntities are entities.

Reply:
It's never inside the tryScoreEntities() function (as does the C# version), I simply moved its location from before tryScoreEntities() to after to be in sync with the original C# implementation.
This PR only tries to make this JS version in sync with the C# one and fix some bugs. If you wish, we can expand the scope on refactoring the overall code structure, but I think it would be better for another PR.